### PR TITLE
Allow multiple terrain (tile) categories

### PR DESCRIPTION
**Pull request by rob-v**
Allow multiple categories for tiles as requested here #13303 to display some tiles in more categories in Map editor.
At the beginning of each list by category are all tiles where first/main category is the selected one and then appended remaining tiles to the list so current order won't be affected.

### CODE REVIEW

File: OpenRA.Game/Map/TileSet.cs [hunk 1]
```diff
@@ -84,7 +84,7 @@ public class TerrainTemplateInfo
 		public readonly int[] Frames;
 		public readonly int2 Size;
 		public readonly bool PickAny;
-		public readonly string Category;
+		public readonly string[] Category;
```
Comment (pchote) [inline OpenRA.Game/Map/TileSet.cs]: Please rename this to `Categories` and an entry to `UpgradeRules.UpgradeTileset` to convert the yaml.

**Comment by rob-v** [comment]
Updated - Category renamed to Categories, tileset .yaml files updated.

**Review by reaperrr — APPROVED** [review]

**Review by GraionDilach — APPROVED** [review]
:+1:

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/11c12ee260168976eff6cd4dc538321768355687...86c6c069481b8da1173eb6df837397ea77b2c11c

**Pull request merged**